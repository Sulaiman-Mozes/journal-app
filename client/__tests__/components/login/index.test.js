import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import LoginComponent from '../../../components/login';


describe('Provider and LoginComponent', () => {
  const props = {
    handleChange: jest.fn(),
    handleSubmit: jest.fn(),
    fields: { email: '', password: '' },
  };
  const loginComponent = shallow(<LoginComponent {...props} />);

  it('renders <Provider/> correctly', () => {
    expect(loginComponent).toMatchSnapshot();
  });
});
