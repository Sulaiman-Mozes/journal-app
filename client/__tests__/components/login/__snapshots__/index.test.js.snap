// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Provider and LoginComponent renders <Provider/> correctly 1`] = `
<div
  className="padding-top form-container padding-bottom"
>
  <form
    className="padding-top text-center border border-light p-5"
    onSubmit={[MockFunction]}
  >
    <p
      className="h4 mb-4"
    >
      Sign in
    </p>
    <input
      className="form-control mb-4"
      id="defaultLoginFormEmail"
      name="email"
      onChange={[MockFunction]}
      placeholder="E-mail"
      required={true}
      type="email"
      value=""
    />
    <input
      className="form-control mb-4"
      id="defaultLoginFormPassword"
      name="password"
      onChange={[MockFunction]}
      placeholder="Password"
      required={true}
      type="password"
      value=""
    />
    <div
      className="d-flex justify-content-around"
    >
      <div>
        <div
          className="custom-control custom-checkbox"
        >
          <input
            className="custom-control-input"
            id="defaultLoginFormRemember"
            type="checkbox"
          />
          <label
            className="custom-control-label"
            htmlFor="defaultLoginFormRemember"
          >
            Remember me 
          </label>
        </div>
      </div>
      <br />
      <div>
        <a
          href="/"
        >
           Â  Forgot password?
        </a>
      </div>
    </div>
    <button
      className="btn btn-info btn-block my-4"
      type="submit"
    >
      Sign in
    </button>
    <p>
      Not a member?
      <NavLink
        to="/signup"
      >
         Register
      </NavLink>
    </p>
    <p>
      or sign in with:
    </p>
    <NavLink
      className="light-blue-text mx-2"
      to="/signup"
    >
      <i
        className="fab fa-facebook-f"
      />
    </NavLink>
    <NavLink
      className="light-blue-text mx-2"
      to="/signup"
    >
      <i
        className="fab fa-twitter"
      />
    </NavLink>
    <NavLink
      className="light-blue-text mx-2"
      to="/signup"
    >
      <i
        className="fab fa-google"
      />
    </NavLink>
  </form>
</div>
`;
